Source/WebCore/page/ResizeObserver.cpp

```diff
@@ -141,6 +143,8 @@ void ResizeObserver::deliverObservations()
     m_activeObservations.clear();
     auto activeObservationTargets = std::exchange(m_activeObservationTargets, { });
 
+    std::exchange(m_targetsWaitingForFirstObservation, { });
```

Comment:
I'm not an expert in these things, but my worry is that, unlike `IntersectionObserver::notify`, you might not be keeping the old value alive by storing it into a variable.

Could it then be possible for the callbacks to be garbage collected before `m_callback->handleEvent(*this, entries, *this);`? I will defer the review to someone who knows better about these things.

Reply:
Yeah, we should clear this after calling m_callback->handleEvent. Maybe use ScopeExit?